qbraid/runtime/azure/provider.py

```diff
@@ -0,0 +1,494 @@
+        )
+
+        quantum_headers = {
+            "Content-type": "application/json",
+            "Authorization": f"Bearer {quantum_access_token}",
+        }
+
+        payload = {"containerName": containerName}
+
+        response = session.post(url, json=payload, headers=quantum_headers)
+        containerSasUri = response.json()["sasUri"]
+
+        payload["blobName"] = "inputData"
+        response = session.post(url, json=payload, headers=quantum_headers)
+        inputSasUri = response.json()["sasUri"]
+
+        payload["blobName"] = "rawOutputData"
+        response = session.post(url, json=payload, headers=quantum_headers)
+        outputSasUri = response.json()["sasUri"]
+
+        return [containerSasUri, inputSasUri, outputSasUri, quantum_headers]
+
+    # pylint: disable=too-many-arguments
+    @staticmethod
+    def create_payload(
+        container_uri: str,
+        input_uri: str,
+        output_uri: str,
+        job_id: str,
+        job_name: str,
+        input_data_format: str,
+        provider: str,
+        backend: str,
+        num_qubits: str,
+        num_shots: int,
+        total_count: int,
+        output_data_format: str,
+    ) -> dict:
+        """Create the payload for the job."""
+        payload = {
+            "containerUri": container_uri,
+            "id": job_id,
+            "inputDataFormat": input_data_format,  # "honeywell.openqasm.v1"
+            "itemType": "Job",
+            "name": job_name,
+            "providerId": provider,
+            "target": backend,  # changeable
+            "metadata": {
+                "qiskit": "True",
+                "name": job_name,
+                "num_qubits": num_qubits,
+                "metadata": "null",
+                "meas_map": "[0]",
+            },
+            "inputDataUri": input_uri,
+            "inputParams": {"shots": num_shots, "count": total_count},
+            "outputDataFormat": output_data_format,  # "honeywell.quantum-results.v1"
+            "outputDataUri": output_uri,
+        }
+
+        return payload
+
+    # pylint: disable=too-many-arguments
+    @staticmethod
+    def submit_job(
+        job_id: str,
+        payload: dict,
+        job_header: dict,
+        location_name: str,
+        subscription_id: str,
+        resource_group_name: str,
+        workspace_name: str,
+        session: Any,
+    ) -> dict:
+        """Submit the job to the Azure Quantum API."""
+        url = (
+            f"https://{location_name}.quantum.azure.com/subscriptions/{subscription_id}"
+            f"/resourceGroups/{resource_group_name}/providers/Microsoft.Quantum/workspaces/"
+            f"{workspace_name}/jobs/{job_id}?api-version=2022-09-12-preview"
+        )
+
+        response = session.put(url, json=payload, headers=job_header)
+        return response.json()
+
+    @staticmethod
+    def get_output_data(container_name: str, connection_string: str) -> dict[str, Any]:
```

Comment:
This function probably belongs in a job class. See: https://docs.qbraid.com/sdk/user-guide/runtime/components#building-pipelines

> [QuantumProvider](https://docs.qbraid.com/sdk/user-guide/runtime/new-provider#quantumprovider): Converts device metadata into accessible Python objects, facilitating interaction with diverse quantum computing resources.
[QuantumDevice](https://docs.qbraid.com/sdk/user-guide/runtime/new-provider#quantumdevice): Describes the unique parameters and operational settings necessary for executing quantum programs on specific hardware.
[QuantumJob](https://docs.qbraid.com/sdk/user-guide/runtime/new-provider#quantumjob): Represents the transitional states of quantum programs, managing both ongoing and completed quantum computations.
[QuantumJobResult](https://docs.qbraid.com/sdk/user-guide/runtime/new-provider#quantumresult): Captures and structures the output from quantum computations, including histograms and raw data, for analysis and interpretation.

That's the next phase of this project.